rules_daml/daml-build.bzl

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2020 The DAML Authors. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+def file_of_target(k):
+    [file] = k.files.to_list()
+    return file
+
+_damlc = attr.label(
+    allow_single_file = True,
+    default = Label("//compiler/damlc"),
+    executable = True,
+    cfg = "host",
+)
+
+def _daml_build_impl(ctx):
+    name = ctx.label.name
+    config = ctx.file.config
+    damls = ctx.files.damls
+    dd = ctx.attr.dar_dict
+    damlc = ctx.file._damlc
+    input_dars = [file_of_target(k) for k in dd.keys()]
+    output_dar = ctx.actions.declare_file(name + ".dar")
+    tmpdir = ctx.actions.declare_directory(name + ".tmp")
```

Comment:
@aherrmann-da Should this be using `declare_directory` or does this run into the same problems since the name is fixed and we need to use `mktmp` in the command itself?

Reply:
The `tmpdir` doesn't require a `declare_directory`. @nickchapman-da you can just `mktmp` it in the `run_shell` command (and also setup a cleanup trap). `declare_file|directory` are used to tell Bazel about outputs to pass them to other actions. IIUC this `tmpdir` is only used within this action.

Also, if possible I'd refrain from using `declare_directory` specifically. With rules_haskell it's causing mysterious "module not found" errors on Windows.

Reply:
My understanding of the previous problem, was that we were building in the source directory, and this caused problems with the way bazel handle globs. Building in a tmpdir should avoid this problem.

Another issue may be that without sandboxing, the results of one build will persist and be unintentionally available to the next. I've added a rm of the tmpdir before the build to avoid these issues.

I did have a remove of the tmpdir after the build action had run, but that broke on windows:
```ERROR: D:/a/1/s/docs/BUILD.bazel:602:1: TreeArtifact docs/daml-upgrade-example-v1.tmp was not created
```
so I removed the remove!

Reply:
The point is that this is not actually a temp dir, its just another directory created by `declare_directory` which has caused problems in the past.  Using `mktemp` is both simpler and avoids any potential issues in `declare_directory`.

Reply:
@aherrmann I can recode to remove the use of `declare_directory` if you think I should. Or we could just see if the way it's used in this rule is ok...

Reply:
Let’s please just switch to `mktemp` for now.

Reply:
Our comments overlapped

Reply:
ok!